internal/cmd/skupper/listener/kube/listener_generate.go

```diff
@@ -39,12 +36,10 @@ func NewCmdListenerGenerate() *CmdListenerGenerate {
 }
 
 func (cmd *CmdListenerGenerate) NewClient(cobraCommand *cobra.Command, args []string) {
-	cli, err := client.NewClient(cobraCommand.Flag("namespace").Value.String(), cobraCommand.Flag("context").Value.String(), cobraCommand.Flag("kubeconfig").Value.String())
+	cli, err := client.NewGenerateCmdClient(cobraCommand.Flag("namespace").Value.String())
```

Comment:
This is not necessary, I suggest to just keep
cmd.namespace = cli.Namespace in the NewClient method.

If generate methods are not checking with a client it is fine to just not call them.

Reply:
removed function call and only set namespace in NewClient